xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-io/src/main/java/org/xwiki/annotation/io/internal/DefaultIOService.java

```diff
@@ -200,12 +205,12 @@ public Collection<Annotation> getAnnotations(String target) throws IOServiceExce
                 return Collections.emptySet();
             }
             for (BaseObject object : objects) {
-                // if it's not on the required target, ignore it
-                if (object == null || !localTargetId.equals(object.getStringValue(Annotation.TARGET_FIELD))) {
-                    continue;
-                }
                 // use the object number as annotation id
-                result.add(loadAnnotationFromObject(object, deprecatedContext));
+                if (object != null && List.of(localTargetId, "")
```

Comment:
Creating a new List just to check if `object.getStringValue(Annotation.TARGET_FIELD)` is equals to `localTargetId` or empty is not great (for performance and readability IMO).

Reply:
What would be a suitable alternative for you?

Reply:
Actually check "if `object.getStringValue(Annotation.TARGET_FIELD)` is equals to `localTargetId` or empty ".

Reply:
ok, I find it more verbose and less readable but ymmv. As I'm fine with both options anyway, I'll change that.